src/resolvers/user.ts

```diff
@@ -196,27 +196,35 @@ export const resolvers: Resolvers = {
         }
 
         const userEmails = await UserEmail.findByUserId(ref, context, context.token.id);
-
         const existing = userEmails.find((entry) => { return entry.email === email });
-        const oldPrimary = userEmails.find((entry) => { return entry.isPrimary === true });
+        const originalState = { ...existing };
+        const oldPrimary = userEmails.find((entry) => { return Boolean(entry.isPrimary) === true });
+
         if (!existing) {
           throw NotFoundError();
         }
 
-        if (oldPrimary) {
-          oldPrimary.isPrimary = false;
-          await new UserEmail(oldPrimary).update(context);
-        }
-
         existing.isPrimary = true;
         const updated = await new UserEmail(existing).update(context);
         if (updated && (!updated.errors || (Array.isArray(updated.errors) && updated.errors.length === 0))) {
+          // Update old primary record to isPrimary = false, if the new one was updated successfullly
```

Comment:
Updated setPrimaryUserEmail to only update oldPrimary if there were no errors when making the set primary update. Also, if there are errors, we should return the original data, not the updated data, with the errors. What was happening was that the user was setting an email to be primary, and the backend response returned data showing that that email's isPrimary was true, even when the change wasn't made. So the page would falsely reflect that update.